src/java/com/sarality/app/view/list/ListViewItemActionPerformer.java

```diff
@@ -0,0 +1,93 @@
+package com.sarality.app.view.list;
+
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+
+import com.sarality.app.view.action.Action;
+import com.sarality.app.view.action.TriggerType;
+
+/**
+ * Performs the Action when an event on an Item in a ListView is triggered.
+ * 
+ * @author abhideep@ (Abhideep Singh)
+ */
+class ListViewItemActionPerformer implements AdapterView.OnItemClickListener, AdapterView.OnItemLongClickListener,
+    View.OnClickListener, View.OnLongClickListener {
+
+  private final static String LIST_POSITION_TAG = "ListPosition";
+  private final static String LIST_ROW_ID_TAG = "ListRowId";
+
+  private final Action<ListViewActionContext> action;
+  private final LinearLayout listView;
+
+  /**
+   * Constructor.
+   *
+   * @param action Action that needs to be performed on ListView item click.
+   */
+  ListViewItemActionPerformer(Action<ListViewActionContext> action) {
+    this.action = action;
+    this.listView = null;
+  }
+
+  /**
+   * Constructor for NonScrollingListView.
+   *
+   * @param action Action that needs to be performed on ListView item click.
+   * @param listView LinearLayout that in used a ListView.
+   */
+  ListViewItemActionPerformer(Action<ListViewActionContext> action, LinearLayout listView) {
+    this.action = action;
+    this.listView = listView;
+  }
+
+  void setup(ListView view, TriggerType triggerType) {
+    if (triggerType == TriggerType.CLICK_LIST_ITEM) {
+      view.setOnItemClickListener(this);
+    } else if (triggerType == TriggerType.LONG_CLICK_LIST_ITEM) {
+      view.setOnItemLongClickListener(this);
+    }
+  }
+
+  void setup(View view, TriggerType triggerType, int position, long rowId) {
+    if (triggerType == TriggerType.CLICK_LIST_ITEM) {
+      view.setOnClickListener(this);
+      view.setTag(LIST_POSITION_TAG.hashCode(), position);
+      view.setTag(LIST_ROW_ID_TAG.hashCode(), rowId);
+    } else if (triggerType == TriggerType.LONG_CLICK_LIST_ITEM) {
+      view.setOnLongClickListener(this);
+      view.setTag(LIST_POSITION_TAG.hashCode(), position);
```

Comment:
Had forgotten about that. We have two ways around the problem.

1) Convert all of appblocks into a an android library projects and then add a resource file to the mix. Ideally we should break it into multiple library projects for view, actions, lists, dialogs etc etc and then include them in the main project. Unfortunately this is a lot of effort.

2) We declare the resource in the flights resource and pass it in as variable to the ActionPerformer.

I'd like to go with 2) with a TODO to move to 1) later. Is that OK?